core/src/main/java/hudson/model/Computer.java

```diff
@@ -604,7 +604,11 @@ public Node getNode() {
 
     @NonNull
     private Node getNodeOrDie() {
-        return nodeName == null ? Jenkins.get() : Jenkins.get().getNode(nodeName);
+        var node = nodeName == null ? Jenkins.get() : Jenkins.get().getNode(nodeName);
+        if (node == null) {
+            throw new IllegalStateException("Can't set a temporary offline cause if the node has been removed");
```

Comment:
I think this nullness check needs to be moved up into the callers, since some are just reading the status, not trying to set it, right?